### DISCUSSION TITLE
Carpet Mod features

### DISCUSSION THREAD

**Discussion by joex92** [PurpurMC/Purpur#1239] [Feature Requests]
add Carpet Mod [settings](https://github.com/gnembon/fabric-carpet/wiki/Current-Available-Settings#carpet-mod-settings), if not all at least most or some of them.
the ones I'm interested are these ones:
- [movableBlockEntities](https://github.com/gnembon/fabric-carpet/wiki/Current-Available-Settings#movableblockentities)
- [movableAmethyst](https://github.com/gnembon/fabric-carpet/wiki/Current-Available-Settings#movableamethyst)
- [pushLimit](https://github.com/gnembon/fabric-carpet/wiki/Current-Available-Settings#pushlimit)
- [quasiConnectivity](https://github.com/gnembon/fabric-carpet/wiki/Current-Available-Settings#quasiconnectivity)
- [railPowerLimit](https://github.com/gnembon/fabric-carpet/wiki/Current-Available-Settings#railpowerlimit)
- [renewableBlackstone](https://github.com/gnembon/fabric-carpet/wiki/Current-Available-Settings#renewableblackstone)
- [sculkSensorRange](https://github.com/gnembon/fabric-carpet/wiki/Current-Available-Settings#sculksensorrange)
- [stackableShulkerBoxes](https://github.com/gnembon/fabric-carpet/wiki/Current-Available-Settings#stackableshulkerboxes)
- [fastRedstoneDust](https://github.com/gnembon/fabric-carpet/wiki/Current-Available-Settings#fastredstonedust)
- [lagFreeSpawning](https://github.com/gnembon/fabric-carpet/wiki/Current-Available-Settings#lagfreespawning)
- [maxEntityCollisions](https://github.com/gnembon/fabric-carpet/wiki/Current-Available-Settings#maxentitycollisions)
- [optimizedTNT](https://github.com/gnembon/fabric-carpet/wiki/Current-Available-Settings#optimizedtnt)
- [xpFromExplosions](https://github.com/gnembon/fabric-carpet/wiki/Current-Available-Settings#xpfromexplosions)
- [xpNoCooldown](https://github.com/gnembon/fabric-carpet/wiki/Current-Available-Settings#xpnocooldown)
- [commandProfile](https://github.com/gnembon/fabric-carpet/wiki/Current-Available-Settings#commandprofile)
- [persistentParrots](https://github.com/gnembon/fabric-carpet/wiki/Current-Available-Settings#persistentparrots)

You can check Carpet Mod's [current settings](https://github.com/gnembon/fabric-carpet/wiki/Current-Available-Settings#carpet-mod-settings) page to see if there are others that might be good...

**Comment by encode42** [comment]
As discussed in the Discord server, movable tile entities are not practically possible in conjunction with the Bukkit API.

Persistent parrots, fast redstone, optimized TNT, lag free spawning, and max entity collisions are already implemented in Paper/Pufferfish/Spigot.

Rail power limit, exp pickup delay, and piston push limit are already implemented in Purpur.

The profiler already exists in a much better form as Spark.

However, the rest of the options seem do-able

**Reply by joex92** [reply]
oh ok...
I haven't gone through every single feature on Purpur just some of them, my bad, but thanks for confirming...

so for movable tile entities to be a thing then Bukkit should add it in their API?

**Reply by encode42** [reply]
It's something that'd break almost every plugin that relies on tile entities being in one place, which isn't something anyone in the fork chain would like to do atm

Maybe in the far future, if Paper hard forks

**Reply by joex92** [reply]
🤔 oh ok...
Then for now, Movable Tile Entities, are impossible to implement...

ok then, I'll look forward for the other features that can be implemented...

thanks.
